Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "SourceAssetsStorage.h"
+#include "RobotImporterUtils.h"
+
+namespace ROS2::Utils
+{
+    /// Function computes CRC32 on first kilobyte of file.
+    AZ::Crc32 GetFileCRC(const AZStd::string& filename)
+    {
+        auto fileSize = AZ::IO::SystemFile::Length(filename.c_str());
+        fileSize = AZStd::min(fileSize, 1024ull); // limit crc computation to first kilobyte
+        if (fileSize == 0)
+        {
+            return AZ::Crc32();
+        }
+        AZStd::vector<char> buffer(fileSize + 1);
+        buffer[fileSize] = '\0';
```

Comment:
fyI: for quicker initialization, the `AZStd::vector` class has a `resize_no_construct` extension which doesn't try to initialize all elements of the buffer to 0.
```suggestion
        AZStd::vector<char> buffer;
        buffer.resize_no_constructor(fileSize + 1);
        buffer[fileSize] = '\0';
```

Reply:
Done